src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3850,19 +3850,64 @@ protected override void VisitStatement(BoundStatement statement)
             var (collectionKind, targetElementType) = getCollectionDetails(node, node.Type);
 
             var resultBuilder = ArrayBuilder<VisitResult>.GetInstance(node.Elements.Length);
-            var elementConversionCompletions = ArrayBuilder<Func<TypeWithAnnotations, TypeWithState>>.GetInstance();
+            var elementConversionCompletions = ArrayBuilder<Func<TypeWithAnnotations /*targetElementType*/, TypeSymbol /*targetCollectionType*/, TypeWithState>>.GetInstance();
             foreach (var element in node.Elements)
+            {
+                visitElement(element);
+                resultBuilder.Add(_visitResult);
+            }
+
+            void visitElement(BoundNode element)
```

Comment:
This local function captures
Also please avoid local function in the middle of the method logic (this could be done in a later commit to reduce diff for now)